include/cgimap/output_formatter.hpp

```diff
@@ -40,7 +40,8 @@ enum class action_type {
 
 namespace {
 
-const char* element_type_name(element_type elt) {
+template <typename T = const char *>
+T element_type_name(element_type elt) noexcept {
```

Comment:
True, using std::string_view wouldn't work for functions expecting null terminated C strings. (Well, technically you could call .data() on it, but that's best avoided since in general string_view strings don't have to be null terminated)